apollo-federation/src/merger/merge_input.rs

```diff
@@ -0,0 +1,605 @@
+use std::collections::HashSet;
+
+use apollo_compiler::Node;
+use apollo_compiler::ast::InputValueDefinition;
+use apollo_compiler::collections::IndexMap;
+use apollo_compiler::schema::Component;
+use apollo_compiler::schema::InputObjectType;
+
+use crate::error::CompositionError;
+use crate::error::FederationError;
+use crate::error::SubgraphLocation;
+use crate::merger::hints::HintCode;
+use crate::merger::merge::Merger;
+use crate::merger::merge::Sources;
+use crate::schema::position::InputObjectTypeDefinitionPosition;
+use crate::supergraph::CompositionHint;
+
+impl Merger {
+    #[allow(dead_code)]
+    pub(crate) fn merge_input(
+        &mut self,
+        sources: &Sources<Node<InputObjectType>>,
+        dest: &InputObjectTypeDefinitionPosition,
+    ) -> Result<(), FederationError> {
+        let added = self.add_fields_shallow(sources, dest);
+
+        for (dest_field, subgraph_fields) in added {
+            if let Some(field_def) = subgraph_fields.values().find_map(|f| f.as_ref()) {
```

Comment:
am I missing something or we actually don't merge the input fields? it looks like we are only adding the field here (which was done in `addFieldShallow` in [JS](https://github.com/apollographql/federation/blob/main/composition-js/src/merging/merge.ts#L1305))

Is `mergeInputField` ([link](https://github.com/apollographql/federation/blob/main/composition-js/src/merging/merge.ts#L2699-L2706)) (merging descriptions, directives, default values, etc) already handled in some other part of the code base?